proto/scheduler_worker.proto

```diff
@@ -0,0 +1,11 @@
+syntax = "proto3";
+
+import "./messages.proto";
+
+service Worker{
+  rpc Schedule(scheduler.WorkLoad) returns (scheduler.ScheduleResponse);
+
+  rpc GetWorkerMetric(scheduler.Empty) returns (scheduler.WorkerMetric);
+
+  rpc GetWorkLoadMetrics(scheduler.Empty) returns (scheduler.WorkLoadMetricResponse);
+}
```

Comment:
This is the `Agent` or `Worker` service. The scheduler is still the server and should be able to schedule instances to a given worker and asynchronously receives status updates. This could look like:

```proto
service Worker {
     // Worker registration.
     // A successful registration means the scheduler can start sending
     // a stream of workload instances to be scheduled. 
     rpc Register(WorkerStatus) returns (stream Workload);

     // Get worker and instances status updates.
     // Returns a stream of Status messages.
     rpc GetStatusUpdates(google.protobuf.Empty) returns (stream Status)
}

```